consensus/src/epoch_manager.rs

```diff
@@ -254,9 +254,12 @@ impl EpochManager {
                     vec![1; proposers.len()]
                 };
 
-                // First block (after genesis) is epoch=1, so that is the first epoch we consider. (Genesis is epoch=0)
+                // Genesis is epoch=0
+                // First block (after genesis) is epoch=1, and is the only block in that epoch.
+                // It has no votes, so we skip it unless we are in epoch 1, as otherwise it will
+                // skew leader elections for exclude_round number of rounds.
                 let first_epoch_to_consider = std::cmp::max(
-                    1,
+                    if epoch_state.epoch == 1 { 1 } else { 2 },
```

Comment:
I don’t quite understand why we want to consider epoch=1 round=1 when we are still in epoch=1?
i.e., why can’t we just change the min epoch to consider as 2 rather than 1

Reply:
we do have multiple rounds in epoch 1, they are just short-circuted and removed after we move to epoch 2.
so we should have leader election work normally in epoch 1. we just don't want to take history from epoch 1 into epoch 2 leader election

https://gist.githubusercontent.com/igor-aptos/8127b53add8fd9e9a83d9dc299a4eb92/raw/6f6506f2f6ca8a583ab830e26e3a93a7f9940947/gistfile1.txt

Reply:
this expression is getting ugly, maybe this is simpler?
```
if epoch <= 2 {
  use_history_from_previous_epoch_max_count = 0;
}
```

Reply:
That's not the same. use_history_from_previous_epoch_max_count is 5 by default (in case we have multiple consecutive reconfigs/governance proposals. So we want first_epoch_to_consider to be 2 for epoch = 6 too.

Reply:
hmm as long as after epoch 1 we have more than 20 rounds it should be good enough?

Reply:
I see what you are saying, but even though expression is shorter, I find it much more confusing at what it does.

so I am going to land as is.